source/dub/internal/git.d

```diff
@@ -101,3 +101,35 @@ private string determineVersionWithGitTool(NativePath path)
 
 	return null;
 }
+
+/** Clones a repository into a new directory.
+
+	Params:
+		remote = The (possibly remote) repository to clone from
+		reference = The branch to check out after cloning
+		destination = Repository destination directory
+
+	Returns:
+		Whether the cloning succeeded.
+*/
+bool cloneRepository(string remote, string reference, string destination)
+{
+	import std.process : Pid, spawnProcess, wait;
+
+	Pid command;
+
+	if (!exists(destination)) {
+		command = spawnProcess(["git", "clone", remote, destination]);
+		if (wait(command) != 0) {
+			return false;
+		}
+	}
+
+	command = spawnProcess(["git", "-C", destination, "checkout", reference]);
+	if (wait(command) != 0) {
+		rmdirRecurse(destination);
+		return false;
+	}
```

Comment:
Do you think we should clone recursively here? Currently dub downloaded packages don't support submodules but I think this would be a great place to support them.

Reply:
yes because the dep specified by the git+... url may contain a a sub dependency that specifies a "path" property that points to where git submodule would be cloned.

Reply:
I wouldn't make it mandatory. Saying carefully, there are some people who use submodules that can form a cycle :)

So `repository a` has `repository b` as submodule and vice versa. Such submodules can still be used If I add both, `repository a` and `repository b` as dependencies and don't clone resursively.

Though this can be easely extended with an additional option, like `recursively: true` in `dub.json`.

Reply:
i don't know, I understood the question of webfreak as a suggestion to execute the command to pull submodule inconditionally (as this command would not fail), i.e without changing anything more.

Reply:
`git submodule update --init`? This would be fine; `git submodule update --init --recursive` not.